src/providers/cross-chain-sync.provider.ts

```diff
@@ -0,0 +1,322 @@
+        network => networksSyncState.value[network] === NetworkSyncState.Synced
+      ) as Network[],
+      unsynced: networksWithValidSyncState.filter(
+        network =>
+          networksSyncState.value[network] === NetworkSyncState.Unsynced
+      ),
+      syncing: networksWithValidSyncState.filter(
+        network =>
+          networksSyncState.value[network] === NetworkSyncState.Syncing ||
+          tempSyncingNetworks.value[account.value]?.networks.includes(network)
+      ),
+    };
+  });
+
+  const hasError = computed(() => {
+    const hasVotingEscrowError = veBalSyncSupportedNetworks.some(network => {
+      return crossChainNetworks[network].isError.value;
+    });
+    return isOmniEscrowError.value || hasVotingEscrowError;
+  });
+
+  const l2VeBalBalances = computed<L2VeBalBalances>(() => {
+    const result = veBalSyncSupportedNetworks.reduce((acc, network) => {
+      acc[network] = crossChainNetworks[network].calculateVeBAlBalance();
+      return acc;
+    }, {});
+    return result;
+  });
+
+  const warningMessage = computed(() => {
+    if (networksBySyncState.value.syncing.length > 0) {
+      return {
+        title: 'Wait until sync finalizes before restaking on L2',
+        text: 'Your veBAL is currently syncing to other networks. Wait until the sync is complete before re-staking any L2 positions in order for your new boosts to apply.',
+      };
+    }
+
+    return {
+      title: '',
+      text: '',
+    };
+  });
+
+  // TODO: add info message
+  const infoMessage = computed(() => {
+    return {
+      title: '',
+      text: '',
+    };
+  });
+
+  async function sync(network: Network) {
+    const contractAddress = configService.network.addresses.omniVotingEscrow;
+    if (!contractAddress) throw new Error('No contract address found');
+    const signer = getSigner();
+    const omniVotingEscrowContract = new OmniVotingEscrow(contractAddress);
+
+    const layerZeroChainId = configs[network].layerZeroChainId;
+
+    if (!layerZeroChainId) {
+      throw new Error('Must specify layer zero chain id');
+    }
+
+    const tx = await omniVotingEscrowContract.estimateSendUserBalance(
+      signer,
+      layerZeroChainId
+    );
+
+    const { nativeFee } = tx;
+    console.log('nativeFee', nativeFee);
+
+    const sendUserBalanceTx = await omniVotingEscrowContract.sendUserBalance({
+      signer,
+      userAddress: account.value,
+      chainId: layerZeroChainId,
+      nativeFee,
+    });
+
+    return sendUserBalanceTx;
+  }
+
+  async function refetch() {
+    await Promise.all([
+      refetchOmniEscrow(),
+      mainnetCrossChainNetwork.refetch(),
+    ]);
+    if (omniEscrowLocksMap.value) {
+      const promises = networksBySyncState.value.syncing.map(networkId => {
+        return crossChainNetworks[networkId].refetch();
+      });
+      await Promise.all(promises);
+    }
+  }
+
+  let disposeRefetchOnInterval: NodeJS.Timeout;
+  function refetchOnInterval() {
```

Comment:
Can you not use a `useIntervalFn` from `vueuse`?

https://github.com/balancer/frontend-v2/blob/develop/src/pages/pool/add-liquidity.vue#L45

Reply:
I think it's better not to use hooks in functions?